EIPS/eip-7577.md

```diff
@@ -0,0 +1,107 @@
+
+## Specification
+
+Once an EIP has moved out of "Draft" status, it MUST use the EIP versioning scheme outlined below. It MAY already use the versioning scheme in "Draft" status, which could be useful if the specification is actively being implemented. If more than one team is implementing the specification, it is RECOMMENDED to change the EIP's status to "Review".
+
+The EIP versioning scheme MUST follow a semantic versioning scheme of `MAJOR.MINOR.PATCH`, which is applied as follows:
+
+1. `MAJOR`: A breaking change to the specifications that requires an implementation change and a change to the reference tests.
+2. `MINOR`: An addition to the specifications that does not require changing an existing implementation, but requires additional implementation and additional test coverage to be added to the reference tests.
+3. `PATCH`: Any cosmetic change to, or a reformulation of, the EIP without specification change.
+
+Before the EIP has moved out of Draft status and is being versioned, the version number MUST initially have `MAJOR` version `0`.
+
+For every change made to an EIP via a PR made to ethereum/EIPs, the PR author MUST add a new entry to the CHANGELOG section of the EIP that outlines the changes made within the PR. This CHANGELOG entry MUST include the following:
+
+1. A new version number that follows the semantic versioning scheme outlined above.
+2. The date when the changes where introduced.
+3. A link to the ethereum/EIPs PR that implements the changes.
+4. A line for each change made to the EIP's specifications that includes a short description of the change.
+
+Additionally, the new version MUST be added to the metadata header of the EIP's markdown file (to a new "version" field), so that it may be easily parsed.
+
+Tooling SHOULD be added to the ethereum/EIPs repository that checks that the CHANGELOG is updated and that the version number in the metadata header matches the latest version number in the CHANGELOG.
+
+EIPS that have dependencies on other EIPS SHOULD include the version of the dependency in the metadata "requires" field.
+
+## Rationale
+
+A semantic versioning scheme enables clearer communication within the community about EIP status and allows the impact of a change to be ascertained at first glance.
+
+Making the version available in the EIP's metadata header additionally allows for programmatic parsing of the version number by tooling used in reference tests or by client teams. Currently, the ethereum/execution-spec-tests repository implements a rudimentary EIP version checker: EIP spec tests are required to declare the EIP's markdown file digest SHA that the test implementation was based on. The current value of the digest SHA is then polled via the Github API to verify that no changes have occurred since the test implementation. While this provides a warning to test implementers that the EIP has changed, it is clearly of limited use.
```

Comment:
Please avoid external links (ethereum/execution-spec-tests).

Reply:
Removed in 92fb8ba7. The repository is still referred to, I hope that's ok?